modules/docs/arrow-docs/docs/docs/patterns/typeclasspolymorphism/README.md

```diff
@@ -0,0 +1,718 @@
+      repository.allTasksByUser(user3).fix().observable.subscribe({ println(it) }, { println(it) })
+    }
+
+    val flowableModule = Module(FlowableK.async())
+    flowableModule.run {
+      repository.allTasksByUser(user1).fix().flowable.subscribe { println(it) }
+      repository.allTasksByUser(user2).fix().flowable.subscribe { println(it) }
+      repository.allTasksByUser(user3).fix().flowable.subscribe({ println(it) }, { println(it) })
+    }
+  }
+}
+```
+
+Printing:
+
+```
+[Task(value=LocalTask assigned to user1)]
+[Task(value=Remote Task assigned to user2)]
+UserNotInRemoteStorage(user=User(userId=UserId(value=unknown user)))
+
+[Task(value=LocalTask assigned to user1)]
+[Task(value=Remote Task assigned to user2)]
+UserNotInRemoteStorage(user=User(userId=UserId(value=unknown user)))
+```
+
+Everything working as expected. 💪
+
+Let's try the [`DeferredK`]({{ '/docs/integrations/kotlinxcoroutines/' | relative_url }}) wrapper for the 
+`kotlinx.coroutines.Deferred` type:
+
+```kotlin
+object test {
+
+  @JvmStatic
+  fun main(args: Array<String>): Unit {
+    val user1 = User(UserId("user1"))
+    val user2 = User(UserId("user2"))
+    val user3 = User(UserId("unknown user"))
+
+    val deferredModule = Module(DeferredK.async())
+    deferredModule.run {
+      runBlocking {
+        try {
+          println(repository.allTasksByUser(user1).fix().deferred.await())
+          println(repository.allTasksByUser(user2).fix().deferred.await())
+          println(repository.allTasksByUser(user3).fix().deferred.await())
+        } catch (e: UserNotInRemoteStorage) {
+          println(e)
+        }
+      }
+    }
+  }
+}
+```
+
+As you know, you're in charge to catch exceptions on coroutines. As you can see, implementation details as this catch, 
+or the observable subscriptions from previous examples, are implementation details related to the given data types being 
+used for each case, so those must live here, in the program's edge.
+
+Same result one more time:
+
+```
+[Task(value=LocalTask assigned to user1)]
+[Task(value=Remote Task assigned to user2)]
+UserNotInRemoteStorage(user=User(userId=UserId(value=unknown user)))
+```
+
+There's also an alternative api provided by Arrow a bit more fancier for [`DeferredK`]({{ '/docs/integrations/kotlinxcoroutines/' | relative_url }}):
+
+```kotlin
+object test {
+
+  @JvmStatic
+  fun main(args: Array<String>): Unit {
+    val user1 = User(UserId("user1"))
+    val user2 = User(UserId("user2"))
+    val user3 = User(UserId("unknown user"))
+
+    val deferredModuleAlt = Module(DeferredK.async())
+    deferredModuleAlt.run {
+      runBlocking {
+        try {
+          println(repository.allTasksByUser(user1).fix().unsafeAttemptSync())
+          println(repository.allTasksByUser(user2).fix().unsafeAttemptSync())
+          println(repository.allTasksByUser(user3).fix().unsafeAttemptSync())
+        } catch (e: UserNotInRemoteStorage) {
+          println(e)
+        }
```

Comment:
This snippet isn't updated to the new api! The intro just works, it just need to be the right code hahah